fvtest/compilertriltest/SimplifierFoldAbsNegTest.cpp

```diff
@@ -0,0 +1,237 @@
+#include "ras/IlVerifierHelpers.hpp"
+#include "SharedVerifiers.hpp" //for NoAndIlVerifier
+
+template <typename T> char* prefixForType();
+template <> char* prefixForType<int32_t>() { return "i"; }
+template <> char* prefixForType<int64_t>() { return "l"; }
+template <> char* prefixForType<  float>() { return "f"; }
+template <> char* prefixForType< double>() { return "d"; }
+
+template <typename T> char* nameForType();
+template <> char* nameForType<int32_t>() { return "Int32" ; }
+template <> char* nameForType<int64_t>() { return "Int64" ; }
+template <> char* nameForType<  float>() { return "Float" ; }
+template <> char* nameForType< double>() { return "Double"; }
+
+std::vector<int32_t> iTestData = { 0, 1, 2, -1, -2, 99999, -99999, std::numeric_limits<int32_t>::max(), std::numeric_limits<int32_t>::min() };
+std::vector<int64_t> lTestData = { 0, 1, 2, -1, -2, 99999, -99999, std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::min() };
+std::vector<  float> fTestData = { 0, 1, 2, -1, -2, 3.14, -3.14, std::numeric_limits<float>::min(), std::numeric_limits<float>::min() };
+std::vector< double> dTestData = { 0, 1, 2, -1, -2, 3.14, -3.14, std::numeric_limits<double>::min(), std::numeric_limits<double>::min() };
+
+template <typename T> std::vector<T> dataForType();
+template <> std::vector<int32_t> dataForType<int32_t>() { return iTestData; }
+template <> std::vector<int64_t> dataForType<int64_t>() { return lTestData; }
+template <> std::vector<  float> dataForType<  float>() { return fTestData; }
+template <> std::vector< double> dataForType< double>() { return dTestData; }
+
+class SimplifierFoldAbsNegTestIlVerifierBase : public TR::IlVerifier
+   {
+   public:
+   int32_t verify(TR::ResolvedMethodSymbol *sym)
+      {
+      for(TR::PreorderNodeIterator iter(sym->getFirstTreeTop(), sym->comp()); iter.currentTree(); ++iter)
+         {
+         int32_t rtn = verifyNode(iter.currentNode());
+         if(rtn)
+            return rtn;
+         }
+      return 0;
+      }
+   protected:
+      virtual int32_t verifyNode(TR::Node *node) = 0;
+   };
+
+/**
+ * Test Fixture for SimplifierFoldAbsNegTest that
+ * selects only the relevant opts for the test case
+ */
+template <typename T>
+class SimplifierFoldAbsNegTest : public TRTest::JitOptTest
+   {
+
+   public:
+   SimplifierFoldAbsNegTest()
+      {
+      /* Add an optimization.
+       * You can add as many optimizations as you need, in order,
+       * using `addOptimization`, or add a group using
+       * `addOptimizations(omrCompilationStrategies[warm])`.
+       * This could also be done in test cases themselves.
+       */
+      addOptimization(OMR::treeSimplification);
+      }
+
+   };
+
+using Types = ::testing::Types<int32_t, int64_t, float, double>;
+TYPED_TEST_CASE(SimplifierFoldAbsNegTest, Types);
+
+
+class NoAbsAbsIlVerifier : public SimplifierFoldAbsNegTestIlVerifierBase
+   {
+   protected:
+   virtual int32_t verifyNode(TR::Node *node) override
```

Comment:
I would love to see more uses of `override` in our code. Unfortunately, it is also one of those C++11 features that is not supported by all our build compilers. (Also, as a general comment, you should only have [exactly one of `virtual` or `override`](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-override), not both :wink: )

Reply:
I see, let me remove the `override` keyword.

> Also, as a general comment, you should only have exactly one of virtual or override, not both

It is an interesting topic, examples from MSDN use both https://msdn.microsoft.com/en-us/library/jj678987.aspx

Reply:
I've updated these functions not using `override` keywords.